src/Compilers/Core/Portable/Syntax/AbstractWarningStateMap.cs

```diff
@@ -50,7 +50,7 @@ public ReportDiagnostic GetWarningState(string id, int position)
         private WarningStateMapEntry GetEntryAtOrBeforePosition(int position)
         {
             Debug.Assert(_warningStateMapEntries != null && _warningStateMapEntries.Length > 0);
-            int r = Array.BinarySearch(_warningStateMapEntries, new WarningStateMapEntry(position));
+            int r = Array.BinarySearch(_warningStateMapEntries, new WarningStateMapEntry(position), new WarningStateMapEntryComparer ());
```

Comment:
You've made a subtle change that will cause this to now box on every comparison. This is actually calling the generic overload - and hence, your comparer should be implement IComparer<T>, not IComparer.